### PR TITLE
MINOR: always set Serde.Long on count operations

### PR DESCRIPTION
**Pull request by dguy**

### CODE REVIEW

**Comment by dguy** [comment]
@guozhangwang, @mjsax, @bbejeck

**Review by guozhangwang — COMMENTED** [review]
LGTM. One general comment: please update the javadoc in KGroupedStream / etc mentioning that Serde.Long would always be used and whatever user specified will be ignored.

**Comment by guozhangwang** [comment]
Please feel free to merge afterwards.

**Review by bbejeck — APPROVED** [review]
Thanks, @dguy.  LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java [hunk 1]
```diff
@@ -265,6 +265,8 @@ private void determineIsQueryable(final String queryableStoreName) {
 
     @Override
     public KTable<K, Long> count(final Materialized<K, Long, KeyValueStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(materialized, "materialized can't be null");
+        materialized.withValueSerde(Serdes.Long());
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java]: Do we really want to always set like this? What's if a user want to provide a custom Long-Serde that just works fine? Maybe we should only overwrite if value serde is `null` ? It's just a thought -- not sure about it.

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java]: That's a good point. I'd agree with you.

Comment (dguy) [reply streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java]: Thanks. Agreed and updated

**Review by mjsax — APPROVED** [review]
LGTM.

**Review by ijuma — APPROVED** [review]
LGTM

**Pull request closed without merging**